backend/routes/teamspace.js

```diff
@@ -368,6 +593,102 @@
 			});
 	}
 
+	function getTeamspaceSettings(req, res, next) {
+		TeamspaceSettings.getTeamspaceSettings(req.params.account).then((settings) => {
+			responseCodes.respond(utils.APIInfo(req), req, res, next, responseCodes.OK, settings);
+		}).catch(err => {
+			responseCodes.respond(utils.APIInfo(req), req, res, next, err, err);
+		});
+	}
+
+	function updateTeamspaceSettings(req, res, next) {
+		TeamspaceSettings.update(req.params.account, req.body).then((settings) => {
+			responseCodes.respond(utils.APIInfo(req), req, res, next, responseCodes.OK, settings);
+		}).catch(err => {
+			responseCodes.respond(utils.APIInfo(req), req, res, next, err, err);
+		});
+	}
+
+	function getMitigationsFile(req, res, next) {
+		FileRef.getMitigationsFile(req.params.account).then((mitigations) => {
+			const timestamp = (new Date()).toLocaleString();
+			const filenamePrefix = (req.params.account + "_" + timestamp + "_").replace(/\W+/g, "_");
+
+			const headers = {
+				"Content-Disposition": "attachment;filename=" + filenamePrefix + "mitigations.csv",
+				"Content-Type": "text/csv"
+			};
+
+			res.set(headers);
+
+			responseCodes.respond(utils.APIInfo(req), req, res, next, responseCodes.OK, mitigations);
+		}).catch(err => {
+			responseCodes.respond(utils.APIInfo(req), req, res, next, err, err);
+		});
+	}
+
+	function uploadMitigationsFile(req, res, next) {
+		const place = utils.APIInfo(req);
+		const {account} = req.params;
+		const sessionId = req.headers[C.HEADER_SOCKET_ID];
+		const user = req.session.user.username;
+
+		function fileFilter(fileReq, file, cb) {
```

Comment:
I think we really should stop doing function within function. at least make it an arrow function.

is this a copy and sight change from the model upload? if so we should refactor it a bit and make both share some common code.